sycl/include/CL/sycl/detail/image_impl.hpp

```diff
@@ -376,16 +376,17 @@ class image_impl final : public SYCLMemObjT<AllocatorT> {
     return PI_MEM_TYPE_IMAGE3D;
   }
 
-  RT::PiMemImageDesc getImageDesc() {
+  RT::PiMemImageDesc getImageDesc(bool InitFromHostPtr) {
     RT::PiMemImageDesc Desc;
     Desc.image_type = getImageType();
     Desc.image_width = MRange[0];
     Desc.image_height = Dimensions > 1 ? MRange[1] : 1;
     Desc.image_depth = Dimensions > 2 ? MRange[2] : 1;
     // TODO handle cases with IMAGE1D_ARRAY and IMAGE2D_ARRAY
     Desc.image_array_size = 0;
-    Desc.image_row_pitch = MRowPitch;
-    Desc.image_slice_pitch = MSlicePitch;
+    // Pitches must be 0 if host ptr is not provided.
```

Comment:
I did not understand this change. How are the fields Desc.image_row_pitch and Desc.image_slice_pitch used? What happens when the user is constructing an image without host_ptr but a valid pitch?
Eg with constructor: 
image(image_channel_order order, image_channel_type type, const range<dimensions> &range, const range<dimensions - 1> &pitch, const property_list &propList = {});

Reply:
> How are the fields Desc.image_row_pitch and Desc.image_slice_pitch used?

There are used as arguments to clCreateImage function. 

> What happens when the user is constructing an image without host_ptr but a valid pitch?

For OpenCL device image is created without host ptr and with pitch == 0.
For host device image is created without host ptr and with pitches that user specifies.

Reply:
Read section 5.3.1.2 in OpenCL spec.
Thanks.